### DISCUSSION TITLE
OSQP w/MKL Pardiso Solver

### DISCUSSION THREAD

**Discussion by erikrforseth** [osqp/osqp#578] [Python Interface]
I'm running Python3.9 on Mac Ventura 13.4.1, 3.6 GHz Quad-Core Intel Core i7.
I have a large sparse linear program which I'm able to solve via OSQP using the QDLDL solver. I understand the MKL Pardiso solver may be more appropriate for my problem, but I get an error when I try to use this solver. I've installed MKL via pip and also via the [standard installer](https://www.intel.com/content/www/us/en/developer/tools/oneapi/onemkl.html) -- for which I also ran 

    source /opt/intel/oneapi/setvars.sh intel64

In any case, when I try to run my code, I get the following error

"Intel MKL FATAL ERROR: Cannot load libmkl_intel_thread.2.dylib."

I get the same issue when I try running in Python 3.8, 3.7, etc. I sort of assume there's some issue with the way MKL has been installed or configured, but I'm not sure how to debug it.

**Comment by erikrforseth** [comment]
I seem to have been able to resolve this by using pip to install an older version of the MKL libraries... I tried mkl==2019.0 and now my code runs without errors. So perhaps there is some incompatibility or extra configuration needed to run OSQP with the latest MKL versions on Mac.

**Reply by imciner2** [reply]
Which version of OSQP are you using? The current master branch (corresponding to the 1.0 beta version) has an updated way of loading MKL, since the older way in 0.6 and before was somewhat fragile.
